packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterForm.tsx

```diff
@@ -169,12 +225,77 @@ function GridFilterForm(props: GridFilterFormProps) {
     [currentOperator],
   );
 
+  const FilterFormDeleteIcon = React.useMemo(
```

Comment:
Why to memoize? The styled component won't react to a change in the theme.

Reply:
Because I have to define them inside the FilterForm components to access `rootProps.components.BaseFormControl`. And I anted to avoid recomputing styled components at every render.

Reply:
A solution to avoid `useMemo`, and not define `styled` components inside the GrisFilterForm would be to create these components outside as follow:

```js
const FilterFormColumnInput = ({children, ...props}) => {
  const rootProps = useGridRootProps();
  const Container = styled(rootProps.components.BaseFormControl, { ... })( ... );
  
  return <Container {...props}>{children}</Container>
}
``` 

By the way, `useMemo` does not seems to break the theme update. I modified one of the styles components to make its background color change according to the theme and it works fine. (I agree, colors are ugly :p ). 

```js
const FilterFormColumnInput = React.useMemo(
    () =>
      styled(rootProps.components.BaseFormControl, {
        name: 'MuiDataGrid',
        slot: 'FilterFormColumnInput',
        overridesResolver: (_, styles) => styles.filterFormColumnInput,
      })(({ theme }) => ({ 
		width: 150,
		backgroundColor: theme.palette.mode === 'dark' ? "red" : "green" })
	),
    [rootProps.components.BaseFormControl],
  );
```

![Peek 2022-01-27 11-06](https://user-images.githubusercontent.com/45398769/151337835-812bb2ea-41c4-436a-8399-c2b4fe3486fa.gif)

Reply:
> Because I have to define them inside the FilterForm components to access rootProps.components.BaseFormControl

You can create the styled component normally, then use the `as` prop to pass the component from the slot:

https://github.com/mui-org/material-ui-x/blob/70a372129854b609a1b352167e40ac01c45fdd6c/packages/grid/_modules_/grid/components/menu/GridMenu.tsx#L94-L95

> By the way, useMemo does not seems to break the theme update.

Yeah, it works but this means that it's not really memoizing the result. It's only memoizing the styled component, not the instance. The costly part of emotion - the serialization of the styles - is still being done in each render. In this case, I would avoid making premature optimizations. If performance was the problem a better solution would be https://github.com/emotion-js/emotion/issues/2529 or export `css` from `@mui/material/styles`.